### PR TITLE
add nfdump stat support for NfSen

### PR DESCRIPTION
**Pull request by VVelox**
DO NOT DELETE THIS TEXT

#### Please note

> Please read this information carefully. You can run `./scripts/pre-commit.php` to check your code before submitting.

- [X] Have you followed our [code guidelines?](http://docs.librenms.org/Developing/Code-Guidelines/)

#### Testers

If you would like to test this pull request then please run: `./scripts/github-apply <pr_id>`, i.e `./scripts/github-apply 5926`
After you are done testing, you can remove the changes with `./scripts/github-remove`.  If there are schema changes, you can ask on discord how to revert.

You can now do fun stuff like find top talkers via nfdump. :3

### CODE REVIEW

**Comment by VVelox** [comment]
https://i.imgur.com/R2G8tF0.png

Image of the new stat page for those curious. :)

**Comment by VVelox** [comment]
To tired to deal with it now, but I will start with cleaning up the style stuff tomorrow night.

**Comment by murrant** [comment]
@VVelox no rush.  You should check the Scrutnizer output too, it has a few items that should be cleaned up.

**Comment by VVelox** [comment]
@murrant All good. The codeclimate stuff is just it being crazy about a handful of non-issues.

File: html/includes/functions.inc.php [hunk 1]
```diff
@@ -1685,3 +1685,165 @@ function get_sensor_label_color($sensor)
...
+ *  4 %Y/%W/%u/%H year/week_of_year/day_of_week/hour
+ *  5 %Y/%j       year/day-of-year
+ *  6 %Y/%j/%H    year/day-of-year/hour
+ *  7 %Y-%m-%d    year-month-day
+ *  8 %Y-%m-%d/%H year-month-day/hour
+ */
+function time_to_nfsen_subpath($time)
+{
+    $time=lowest_five_minutes($time);
+    $layout=$config['nfsen_subdirlayout'];
```
Comment (murrant) [inline html/includes/functions.inc.php]: Can you use `Config::get('nfsen_subdirlayout')` here and everywhere?

File: html/includes/functions.inc.php [hunk 2]
```diff
@@ -1685,3 +1685,165 @@ function get_sensor_label_color($sensor)
...
+
+    foreach ($nfsen_rrd_dirs as $nfsen_dir) {
+        $host_dir=$nfsen_dir.'/'.$nfsen_hostname.'/';
+
+        if (is_dir($host_dir)) {
+            $nfsen_RRD_channel_glob=$host_dir.'*.rrd';
+            foreach (glob($nfsen_RRD_channel_glob) as $nfsen_RRD) {
+                $channel = str_replace(array($host_dir, '.rrd'), '', $nfsen_RRD);
+
+                $channels[$channel]=$nfsen_dir.$channels.'.rrd';
```
Comment (murrant) [inline html/includes/functions.inc.php]: $channels is an string as defined above, but you use it as an array here.

**Comment by VVelox** [comment]
@murrant Resolved and merged... it is all now good. :)

**Comment by VVelox** [comment]
@murrant Hey. I think it merged all sanely now.

**Comment by VVelox** [comment]
@murrant Hmmm... appears to not have sanely merged. Just doing a rebuild of my dev enviroment and this one is proving problematic.

**Comment by VVelox** [comment]
@murrant Replaced with https://github.com/librenms/librenms/pull/10376 . Tested against the current master and it works wonderfully.

**Pull request closed without merging**